### PR TITLE
feat: multiplication of rational number

### PR DESCRIPTION
**Pull request by Roznovjak**
I would like to use it in trade limit calculations to calculate some percentage of a number. It's quite common operation, might also come handy in the future.

### CODE REVIEW

**Review by enthusiastmartin — COMMENTED** [review]
I'd consider to look at @apopiak Ratio implementation and use that.( in  #66 ) 

We probably want to have one type representing rational number and have all the math logic implemented for that. 

that way it wont be confusing what to use and how.

File: src/fee.rs [hunk 1]
```diff
@@ -18,6 +19,14 @@ pub fn calculate_pool_trade_fee(amount: Balance, fee: (u32, u32)) -> Option<Bala
         .checked_mul(numerator as Balance)
 }
 
+pub fn multiply_rational(numerator: u128, denominator: u128, amount: Balance) -> Option<Balance> {
+    let (numerator, denominator, amount) = to_u256!(numerator, denominator, amount);
+
+    let result = amount.checked_mul(numerator).and_then(|v| v.checked_div(denominator))?;
+
+    to_balance!(result).ok()
+}
```
Comment (apopiak) [inline src/fee.rs:28]: it's very similar to [`multiply_by_rational_with_rounding`](https://paritytech.github.io/substrate/master/sp_arithmetic/helpers_128bit/fn.multiply_by_rational_with_rounding.html), do we want a new function for that?

Comment (enthusiastmartin) [reply src/fee.rs:28]: i would try to avoid if possible.  I would prefer to consolidate and have one way.

Comment (Roznovjak) [reply src/fee.rs:28]: Just to make it clear, do we want to implement it for `Ratio` or use `multiply_by_rational_with_rounding` instead? I suggest implement a new method for `Ratio` and use `multiply_by_rational_with_rounding` in the implementation of this method.

Comment (apopiak) [reply src/fee.rs:28]: sure we can implement it on `Ratio`
(would do it in a new PR, though)

**Pull request closed without merging**